src/daemon/rpmostree-sysroot-upgrader.c

```diff
@@ -548,6 +550,9 @@ generate_treespec (RpmOstreeSysrootUpgrader *self)
                                   self->override_remove_packages->len);
     }
 
+  if (self->flags & RPMOSTREE_SYSROOT_UPGRADER_FLAGS_PKGCACHE_ONLY)
+    g_key_file_set_boolean (treespec, "tree", "cache-only", TRUE);
```

Comment:
Hmm...I don't think of this as a treespec thing, it's more of a transient config flag, like `rpmostree_core_set_cache_only()`.

Reply:
Yeah, I actually had it as a flag at first. But it needs to affect how `rpmostree_context_setup()`. I suppose we can just enforce that it gets called before calling `setup()` for it to have any effect.